File: ch21p641editInsertAndRearrangeRows/p543p557editInsertAndRearrangeRows/MyCell.m

#import "MyCell.h"


@implementation MyCell


- (void) didTransitionToState:(UITableViewCellStateMask)state {
    [super didTransitionToState:state];
    if (state == UITableViewCellStateEditingMask) {
        self.textField.enabled = YES;
    }
    if (state == UITableViewCellStateDefaultMask) {
        self.textField.enabled = NO;
    }
}

@end

File: iOS9bookExamples/bk2ch08p447editInsertAndRearrangeRows/ch21p741editInsertAndRearrangeRows/MyCell.swift

import UIKit

class MyCell : UITableViewCell {
    @IBOutlet weak var textField : UITextField!
    override func didTransitionToState(state: UITableViewCellStateMask) {
        self.textField.enabled = state.contains(.ShowingEditControlMask)
        super.didTransitionToState(state)
    }
}
